pkg/manager/manager_utils.go

```
@@ -112,30 +214,390 @@ func isGitHubTreeURL(ref string) bool {
+// compileGlob compiles a glob pattern to use for matching.
+func compileGlob(pattern string) (glob.Glob, error) {
+	return glob.Compile(pattern, '/')
+}
+
+// matchGlob matches a path against a compiled glob pattern.
+func matchGlob(g glob.Glob, path string) bool {
+	return g.Match(path)
+}
+
 // generateRuleKey creates a rule key from a reference.
 func generateRuleKey(ref string) string {
```

golangci-lint (1.64.8)

cognitive complexity 37 of func `generateRuleKey` is high (> 20)

(gocognit)